lib/hotel.rb

```diff
@@ -0,0 +1,209 @@
+require 'date'
+require_relative 'reservation'
+require 'awesome_print'
+require 'pry'
+
+module Admin
+  class Hotel
+    attr_reader :rooms
+
+    def initialize(rooms)
+      @rooms = Array.new(rooms)
+
+    end
+
+    def reserve_room(reservation)
+      i = 0
+      reservation_array = []
+      reservation_made = 0
+
+
+      until i == @rooms.length || reservation_made == reservation.number_of_rooms
+
+        if @rooms[i] == nil
+          @rooms[i] = []
```

Comment:
You could clean this up by creating one initial local variable equal to `@rooms[i]` so you don't have to duplicate this everywhere

Reply:
So I would define  @rooms[i] = [] as the default value in the initializer?

Reply:
No, something like `this_room = @rooms[i]` and then use `this_room` wherever you were previously using `@rooms[i]`